internal/testutils/tls_creds.go

```diff
@@ -74,3 +74,25 @@ func CreateServerTLSCredentials(t *testing.T, clientAuth tls.ClientAuthType) cre
 		ClientCAs:    ca,
 	})
 }
+
+func CreateServerTLSCredentialsSPIFFE(t *testing.T, clientAuth tls.ClientAuthType) credentials.TransportCredentials {
```

Comment:
Any reason this is server-side only?

Reply:
It's a test helper for creating intended server-side TLS creds with spiffe test credentials (and takes the form of other similar test helpers in the file)

Reply:
Thanks. What I want to make sure of is that we are testing the SPIFFE verification on both the client and server side of the connection - can you confirm?

Reply:
In this PR, no, but in a followup, yes.
[In this code server-side is not supported](https://github.com/grpc/grpc-go/blob/775150f68c1759ad4b1230d09fe41efb6f3c76ad/internal/xds/bootstrap/tlscreds/bundle.go#L136-L138)

Now, if you ask me why configuring xds tls creds via bundles only supports client-side, I'm not 100% sure, but I was working within the existing code structure.
Digging a little, the original `bundle.go` file that this PR is modifying was originally in the path `xds/internal/xdsclient/tlscreds/bundle.go` - https://github.com/grpc/grpc-go/commit/6bc19068a7682c6b941ffbbded3740f3728a26e9

Either way, the next PR very explicitly adds to the `handshake_info.go` and will test client and server-side